jenkins/verify-chef.sh

```diff
@@ -74,6 +74,12 @@ export FORCE_FFI_YAJL
 PATH=/opt/$PROJECT_NAME/bin:/opt/$PROJECT_NAME/embedded/bin:$PATH
 export PATH
 
-# Test against the appbundle'd Chef
-cd /opt/$PROJECT_NAME/embedded/apps/chef
+# Test against the vendored Chef gem
+cd /opt/$PROJECT_NAME/embedded/lib/ruby/gems/*/gems/chef-[0-9]*
+
+if [ ! -f "Gemfile.lock" ]; then
+  echo "Chef gem does not contain a Gemfile.lock! This is needed to run any tests."
+  exit 1
+fi
+
```

Comment:
We'll need to ensure these changes work on all *nix platforms...worried about more esoteric things like FreeBSD and Solaris. A simple test build should verify things.